src/nodes/explosive.rs

```diff
@@ -0,0 +1,144 @@
+pub mod explosives {
```

Comment:
Why its a mod within a mod? Maybe just remove this line?

Reply:
That's for accessing the `detonation_parameters` struct from elsewhere… Otherwise I would have to add both to nodes.rs, which doesn't seem right. Though this doesn't really seem like the right solution either, especially since every field needs to be marked `pub`.

Reply:
but it will be equally available without a mod within mod?

Reply:
The mod is for grouping, so there aren't any non-node structs in nodes.rs. But it was mostly a temporary solution. Have you got some cleaner solution? Of course, I could just add it straight to nodes.rs but since it is closely tied to the explosives they ought to stick together somehow. Though the mod makes the explosives unnecessarily isolated...

Reply:
In nodes.rs there are only things you explicitly decided to `use`. Therefore if you do not use something in `nodes.rs` - it will not make it there.
If you want a module, not struct being available in `nodes` - do not use anything, do `pub mod explosive`.

Reply:
It seems I overcomplicated things, got it figured out now!